continuous-integration/linux/build.sh

```diff
@@ -2,12 +2,29 @@
 
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 ROOT_DIR=$(realpath "$SCRIPT_DIR/../..")
+BUILD_ROOT_DIR="$ROOT_DIR/build"
 
-cd $ROOT_DIR
-mkdir build || exit $?
-cd build || exit $?
-cmake -DCMAKE_CXX_COMPILER=clang++-8 \
-    -DUSE_EIGEN3=OFF \
-    -DUSE_OPENCV=OFF \
-    .. || exit $?
-cmake --build . || exit $?
+function build()
+{
+    COMPILER=$1
+    BUILD_SUBDIR=$2
+    BUILD_DIR="$BUILD_ROOT_DIR/$BUILD_SUBDIR"
+
+    echo "-----------------------------------------------"
+    echo "            Building with $COMPILER"
+    echo "-----------------------------------------------"
+
+    cd $ROOT_DIR || exit $?
+    mkdir -p $BUILD_DIR || exit $?
+    cd $BUILD_DIR || exit $?
```

Comment:
You can also do:
`mkdir --parents "$ROOT_DIR/$BUILD_DIR" || exit $?`
You should also use "" around variables containing paths, since there could be a space in one of them.

Reply:
Good idea to use `--parents` instead of `-p`. Also removed the unnecessary `cd $ROOT_DIR`. Added quotes too.